### PR TITLE
Make boskos client thread safe

### PR DESCRIPTION
**Pull request by krzyzacy**
ref https://github.com/kubernetes/test-infra/pull/2655#discussion_r114946346

/assign @fejta

### CODE REVIEW

**Comment by krzyzacy** [comment]
and ref #2666

File: boskos/client/client.go [hunk 1]
```diff
@@ -31,6 +32,7 @@ type Client struct {
 	url       string
 	resources []string
 	owner     string
+	lock      sync.Mutex
```
Comment (mikedanese) [inline boskos/client/client.go]: lock should be above the fields it guards. If it guards all fields it should be embedded. if it guards a non-obvious subset, subset should be documented with a comment.

Comment (krzyzacy) [inline boskos/client/client.go]: Thanks for pointing! Moved up!

**Comment by fejta** [comment]
/assign @mikedanese 

Want to look? You probably understand correct go concurrency patterns than me.

**Comment by krzyzacy** [comment]
@mikedanese thanks!

**Pull request merged**